src/traffic_cache_tool/CacheTool.cc

```diff
@@ -255,6 +255,8 @@ class VolumeAllocator
       : _config(config), _size(size), _deficit(deficit), _shares(shares)
     {
     }
+    V(const V &v) : _config(v._config), _size(v._size), _deficit(v._deficit), _shares(v._shares) {}
```

Comment:
Hmm, it seems like this could just be:
```
V(const V &v) = default;
```
But it also seem like you could just let both the copy constructor and the copy assign be defined implicitly.  @SolidWallOfCode did you originally write this code, what's the dealio?

Reply:
Using `= default` and generate copy constructor looks better. Will do.

@SolidWallOfCode will describe better, but if I use the default copy assignment, I got below error.
```
/usr/local/opt/llvm/include/c++/v1/algorithm:3945:17: error: no matching function for call to 'swap'
                swap(*__first, *__last);
                ^~~~
/usr/local/opt/llvm/include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::__sort<(lambda at traffic_cache_tool/CacheTool.cc:407:37) &, ct::VolumeAllocator::V *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/usr/local/opt/llvm/include/c++/v1/algorithm:4158:12: note: in instantiation of function template specialization 'std::__1::sort<ct::VolumeAllocator::V *, (lambda at traffic_cache_tool/CacheTool.cc:407:37) &>' requested here
    _VSTD::sort<_Tp*, _Comp_ref>(__first.base(), __last.base(), __comp);
           ^
traffic_cache_tool/CacheTool.cc:407:8: note: in instantiation of function template specialization 'std::__1::sort<ct::VolumeAllocator::V, (lambda at traffic_cache_tool/CacheTool.cc:407:37)>' requested here
  std::sort(_av.begin(), _av.end(), [](V const &lhs, V const &rhs) { return lhs._deficit > rhs._deficit; });
       ^
```

Reply:
Ah I didn't know an implicit copy assignment isn't defined if any member variables are references.